verilog/CST/declaration.cc

```diff
@@ -140,9 +140,17 @@ const verible::SyntaxTreeNode& GetParenGroupFromModuleInstantiation(
 const verible::SyntaxTreeLeaf&
 GetUnqualifiedIdFromVariableDeclarationAssignment(
     const verible::Symbol& variable_declaration_assign) {
-  return *AutoUnwrapIdentifier(*ABSL_DIE_IF_NULL(
+  const verible::Symbol* identifier = ABSL_DIE_IF_NULL(
       GetSubtreeAsSymbol(variable_declaration_assign,
-                         NodeEnum::kVariableDeclarationAssignment, 0)));
+                         NodeEnum::kVariableDeclarationAssignment, 0));
+  if (identifier->Kind() == verible::SymbolKind::kLeaf) {
```

Comment:
Can you add a test case for this that makes sure it doesn't crash?  It's ok that 'branch' isn't recognized as an identifier for now, just include a short comment about it at the test case.